pkg/storage/chunk/client/baidubce/bos_storage_client.go

```diff
@@ -0,0 +1,171 @@
+package baidubce
+
+import (
+	"context"
+	"flag"
+
+	"io"
+	"time"
+
+	"github.com/baidubce/bce-sdk-go/bce"
+	"github.com/baidubce/bce-sdk-go/services/bos"
+	"github.com/baidubce/bce-sdk-go/services/bos/api"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/instrument"
+
+	"github.com/grafana/loki/pkg/storage/chunk/client"
+)
+
+// NoSuchKeyErr The resource you requested does not exist.
+// refer to: https://cloud.baidu.com/doc/BOS/s/Ajwvysfpl
+//			 https://intl.cloud.baidu.com/doc/BOS/s/Ajwvysfpl-en
+const NoSuchKeyErr = "NoSuchKey"
+
+const DefaultEndpoint = bos.DEFAULT_SERVICE_DOMAIN
+
+var bosRequestDuration = instrument.NewHistogramCollector(prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Namespace: "loki",
+	Name:      "bos_request_duration_seconds",
+	Help:      "Time spent doing bos requests.",
```

Comment:
Can you change this help message without changing tests? The phrase should also include the units, so this change might need to be "Time spent, in seconds, doing BOS requests."

```suggestion
	Help:      "Time spent doing BOS requests.",
```